pyretailscience/plots/histogram.py

```diff
@@ -0,0 +1,256 @@
+        range_upper=range_upper,
+        range_method=range_method,
+    )
+
+    cmap = get_base_cmap()
+
+    num_histograms = _get_num_histograms(df=df, value_col=value_col, group_col=group_col)
+
+    ax = _plot_histogram(
+        df=df,
+        value_col=value_col,
+        group_col=group_col,
+        ax=ax,
+        cmap=cmap,
+        num_histograms=num_histograms,
+        **kwargs,
+    )
+
+    ax = gu.standard_graph_styles(
+        ax=ax,
+        title=title,
+        x_label=x_label,
+        y_label=y_label,
+        legend_title=legend_title,
+        move_legend_outside=move_legend_outside,
+    )
+
+    if source_text:
+        gu.add_source_text(ax=ax, source_text=source_text)
+
+    return gu.standard_tick_styles(ax=ax)
+
+
+def _prepare_value_col(df: pd.DataFrame | pd.Series, value_col: str | list[str] | None) -> list[str]:
+    """Ensures that value_col is properly handled and returned as a list.
+
+    Args:
+        df (pd.DataFrame | pd.Series): The input dataframe or series.
+        value_col (str or list of str, optional): The column(s) to plot. If a single string, it is converted to a list.
+
+    Returns:
+        list[str]: The processed value_col as a list of strings.
+    """
+    if isinstance(df, pd.Series):
+        return ["value"] if value_col is None else [value_col]
+
+    if value_col is None:
+        raise ValueError("Please provide a value column to plot.")
+
+    if isinstance(value_col, str):
+        value_col = [value_col]
+
+    return value_col
+
+
+def apply_range_clipping(
+    df: pd.DataFrame,
+    value_col: list[str],
+    range_lower: float | None,
+    range_upper: float | None,
+    range_method: Literal["clip", "fillna"],
+) -> pd.DataFrame:
+    """Applies range clipping or filling based on the provided method and returns the modified dataframe.
+
+    Args:
+        df (pd.DataFrame): The dataframe to apply range clipping to.
+        value_col (list of str): The column(s) to apply clipping or filling to.
+        range_lower (float, optional): Lower bound for clipping or filling NA values.
+        range_upper (float, optional): Upper bound for clipping or filling NA values.
+        range_method (Literal, optional): Whether to "clip" values outside the range or "fillna". Defaults to "clip".
+
+    Returns:
+        pd.DataFrame: The modified dataframe with the clipping or filling applied.
+    """
+    if range_lower is not None or range_upper is not None:
+        if range_method == "clip":
+            return df.assign(**{col: df[col].clip(lower=range_lower, upper=range_upper) for col in value_col})
+        return df.assign(
+            **{
+                col: df[col].apply(
+                    lambda x: np.nan
+                    if (range_lower is not None and x < range_lower) or (range_upper is not None and x > range_upper)
+                    else x,
+                )
+                for col in value_col
+            },
+        )
+
+    return df
```

Comment:
This flow removes the double negatives and simplifies the code. Not sure about the formatting thought. It's a bit tough to do in github.

```suggestion
    if range_lower is None and range_upper is None:
        return df
        
      if range_method == "clip":
          return df.assign(**{col: df[col].clip(lower=range_lower, upper=range_upper) for col in value_col})
          
      return df.assign(
            **{
                col: df[col].apply(
                    lambda x: np.nan
                    if (range_lower is not None and x < range_lower) or (range_upper is not None and x > range_upper)
                    else x,
                )
                for col in value_col
            },
        )
```